pkg/activator/handler/handler_test.go

```diff
@@ -268,6 +308,159 @@ func TestActivationHandler(t *testing.T) {
+			namespace:   fmt.Sprintf("real-namespace-%d", rev),
+			revName:     "real-name",
+			serviceName: "real-name-service",
+			wantBody:    wantBody,
+			requests:    21,
+			respCh:      respCh,
+			revisions:   revisions,
+		}
+		act := newStubActivator(config.namespace, config.revName, server)
+		handler := getHandler(throttler, t, act)
+		sendRequests(config, handler)
+	}
+	assertResponses(wantedSuccess, wantedFailure, overallRequests, respCh, t)
+}
+
+type Config struct {
+	namespace     string
+	revName       string
+	serviceName   string
+	requests      int
+	respCh        chan *httptest.ResponseRecorder
+	wantBody      string
+	breakerParams queue.BreakerParams
+	revisions     int
+}
+
+func sendRequests(config Config, handler ActivationHandler) {
+	for i := 0; i < config.requests; i++ {
+		go func() {
+			resp := httptest.NewRecorder()
+			req := httptest.NewRequest("POST", "http://example.com", nil)
+			req.Header.Set(activator.RevisionHeaderNamespace, config.namespace)
+			req.Header.Set(activator.RevisionHeaderName, config.revName)
+			handler.ServeHTTP(resp, req)
+			config.respCh <- resp
+		}()
+	}
+}
+
+func getThrottler(breakerParams queue.BreakerParams, t *testing.T) *activator.Throttler {
+	endpointsGetter := func(activator.RevisionID) (int32, error) {
+		return breakerParams.InitialCapacity, nil
+	}
+	throttlerParams := activator.ThrottlerParams{BreakerParams: breakerParams, Logger: TestLogger(t), GetRevision: stubRevisionGetter, GetEndpoints: endpointsGetter}
+	throttler := activator.NewThrottler(throttlerParams)
+	return throttler
+}
+
+func run(conf Config, throttler *activator.Throttler, t *testing.T) {
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			io.WriteString(w, conf.wantBody)
+		}),
+	)
+	act := newStubActivator(conf.namespace, conf.revName, server)
+	handler := getHandler(throttler, t, act)
+	sendRequests(conf, handler)
+}
+
+func getHandler(throttler *activator.Throttler, t *testing.T, act activator.Activator) ActivationHandler {
+	rt := util.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		resp, err := http.DefaultTransport.RoundTrip(r)
+		if err != nil {
+			return resp, err
+		}
+		time.Sleep(50 * time.Millisecond)
+		return resp, nil
+	})
+	handler := ActivationHandler{
+		Activator: act,
+		Transport: rt,
+		Logger:    TestLogger(t),
+		Reporter:  &fakeReporter{},
+		Throttler: throttler,
+	}
+	return handler
+}
+
+func assertResponses(wantedSuccess, wantedFailure, overallRequests int, respCh chan *httptest.ResponseRecorder, t *testing.T) {
+	t.Helper()
+	var succeeded int
+	var failed int
+	wantedOverflowMessage := activator.ErrActivatorOverload.Error() + "\n"
+	for i := 0; i < overallRequests; i++ {
+		resp := <-respCh
+		switch resp.Code {
+		case successCode:
+			gotBody, _ := ioutil.ReadAll(resp.Body)
+			if string(gotBody) != wantBody {
+				t.Errorf("Unexpected response body. Want %q, got %q", wantBody, gotBody)
```

Comment:
got before want is the preferred way. (i.e.g ResponseBody = %q, want: %q)

Reply:
I wouldn't say it is consistent across all the tests and it makes a big difference, but I could change it if you think it is necessary.

Reply:
I'd really appreciate. And I unify that wherever I see it.
Thank you!